python/ray/train/tensorflow/tensorflow_predictor.py

```diff
@@ -21,23 +21,38 @@ class TensorflowPredictor(Predictor):
         preprocessor: A preprocessor used to transform data batches prior
             to prediction.
         model_weights: List of weights to use for the model.
+        use_gpu: If prediction is done on GPU where model will be moved to
+            GPU device upon instantiation.
     """
 
     def __init__(
         self,
         model_definition: Union[Callable[[], tf.keras.Model], Type[tf.keras.Model]],
         preprocessor: Optional["Preprocessor"] = None,
         model_weights: Optional[list] = None,
+        use_gpu: bool = False,
     ):
         self.model_definition = model_definition
         self.model_weights = model_weights
         self.preprocessor = preprocessor
 
+        self.use_gpu = use_gpu
+        # TensorFlow model objects cannot be pickled, therefore we use
+        # a callable that returns the model and initialize it here,
+        # instead of having an initialized model object as an attribute.
+        if use_gpu:
+            # TODO (jiaodong): #26249 Use multiple GPU devices with sharded input
+            with tf.device("GPU:0"):
+                self.model = self.model_definition()
+        else:
+            self.model = self.model_definition()
```

Comment:
model init and config is moved to constructor rather than upon every predict() call.